hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-common/src/main/java/org/apache/hadoop/yarn/server/federation/store/impl/MemoryFederationStateStore.java

```diff
@@ -312,4 +324,41 @@ public Version loadVersion() {
     return null;
   }
 
+  @Override
+  public AddReservationHomeSubClusterResponse addReservationHomeSubCluster(
+      AddReservationHomeSubClusterRequest request) throws YarnException {
+    FederationReservationHomeSubClusterStoreInputValidator
+        .validateAddReservationHomeSubClusterRequest(request);
```

Comment:
These names are pretty verose.
If we are in FederationReservationHomeSubClusterStoreInputValidator isn't enough to say validateRequest? or validate as the type is in the arg?

Reply:
Thanks for your suggestion, I have modified the code, the method is called `validate`